MikuMikuWorld/ScoreContext.cpp

```diff
@@ -987,6 +987,109 @@ namespace MikuMikuWorld
 		pushHistory("Split hold", prev, score);
 	}
 
+	void ScoreContext::repeatMidsInSelection() {
+
+		if (selectedNotes.size() < 3)
+			return;
+
+		Score prev = score;
+
+		Note& note = score.notes[*selectedNotes.begin()];
+		if (!(note.getType() == NoteType::HoldMid || note.getType() == NoteType::Hold))
+			return;
+
+		int startIndex;
+
+		if (note.getType() == NoteType::HoldMid)
+		{
+			startIndex = note.parentID;
+		}
+		else
+		{
+			startIndex = *selectedNotes.begin();
+		}
+		
+		HoldNote& hold = score.holdNotes[startIndex];
+
+		std::vector<int> sortedSelection;
+		sortedSelection.insert(sortedSelection.end(), selectedNotes.begin(), selectedNotes.end());
+		std::sort(sortedSelection.begin(), sortedSelection.end(),
+		          [this](int a, int b) { return score.notes[a].tick < score.notes[b].tick; });
+
+		Note& patternStart = score.notes.at(*sortedSelection.begin());
+		std::reverse(sortedSelection.begin(), sortedSelection.end());
+		Note& patternEnd = score.notes.at(*sortedSelection.begin());
+		std::reverse(sortedSelection.begin(), sortedSelection.end());
+
+		Note& holdStart = score.notes.at(hold.start.ID);
+		Note& holdEnd = score.notes.at(hold.end);
+
+		// score.notes.at(hold.start.ID).tick = 0;
+		// score.notes.at(hold.end).flick = FlickType::Default;
+
+		int patternHeight = patternEnd.tick - patternStart.tick;
+
+		int itterations = std::floor((holdEnd.tick - holdStart.tick) / patternHeight);
+
+		int startPos = findHoldStep(hold, patternStart.ID);
+		int endPos = findHoldStep(hold, patternEnd.ID);
+
+		if (startPos == -1)
+		{
+			hold.steps[endPos].ease = hold.start.ease;
+		}
+		else
+		{
+			hold.steps[endPos].ease = hold.steps[startPos].ease;
+		}
+
+		for (int j = 1; j < sortedSelection.size(); j++)
+		{
+			Note& currentRep = score.notes.at(sortedSelection[j]);
+			int jPos = findHoldStep(hold, currentRep.ID);
+
+			if (j == sortedSelection.size() - 1)
+			{
+				jPos = findHoldStep(hold, score.notes.at(sortedSelection[0]).ID);
+			}
+
+			for (int i = 1; i < itterations; i++)
+			{
+				int lane = std::clamp(currentRep.lane + i * (patternEnd.lane - patternStart.lane),
+				                      0, 12 - currentRep.width);
```

Comment:
Lane extension is not respected. Pass `ScoreContext&` to repeatMidsInSelection? (You can refer `ScoreEditorTimeline::noteControl` for implementation